ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Client.java

```diff
@@ -92,6 +93,30 @@
    */
   CompletableFuture<Void> terminatePushQuery(String queryId);
 
+  /**
+   * Returns the list of ksqlDB streams from the ksqlDB server's metastore.
+   *
+   * @return list of streams
+   */
+  CompletableFuture<List<StreamInfo>> listStreams();
+
+  /**
+   * Returns the list of ksqlDB tables from the ksqlDB server's metastore
+   *
+   * @return list of tables
+   */
+  CompletableFuture<List<TableInfo>> listTables();
+
+  /**
+   * Returns the list of Kafka topics available for use with ksqlDB.
+   *
+   * <p>If a non-200 response is received from the server, the {@code CompletableFuture} will be
+   * failed.
```

Comment:
might be worth adding some documentation as to why this is different than `listStreams` and `listTopic` - maybe (if I'm understand what you're saying correctly):
```suggestion
   * <p>If the ksqlDb server receives a non-200 response is received from the kafka broker, the {@code CompletableFuture} will be
   * failed.
```

Reply:
It's actually `listStreams` and `listTables` that were different from all the rest of the methods (if you look at the other Client methods, you'll see they all have the same note as `listTopics`), since if ksqlDB is operating correctly, those shouldn't ever fail. However, it occurs to me that those requests can still fail if, e.g., ksqlDB is unavailable, or because of misconfigurations in the client, so I've added the note to those methods as well.